src/Definition/Doc.elm

```diff
@@ -0,0 +1,395 @@
+
+                        FoldedSource sources ->
+                            let
+                                viewFoldedSource { isFolded, summary, details } =
+                                    if isFolded then
+                                        div [ class "is-folded" ]
+                                            [ Icon.view Icon.caretRight
+                                            , UI.inlineCode [] (viewSyntax summary)
+                                            ]
+
+                                    else
+                                        div []
+                                            [ Icon.view Icon.caretDown
+                                            , viewSource details
+                                            ]
+                            in
+                            div [ class "folded-sources" ]
+                                (List.map viewFoldedSource sources)
+
+                        Example syntax ->
+                            span [ class "example-inline" ] [ UI.inlineCode [] (viewSyntax syntax) ]
+
+                        ExampleBlock syntax ->
+                            div [ class "example" ] [ UI.codeBlock [] (viewSyntax syntax) ]
+
+                        Link ref ->
+                            a [ onClick (refToMsg ref) ] [ text (Reference.toHumanString ref) ]
+
+                        Signature signatures ->
+                            div [ class "signatures" ] (List.map (\( _, s ) -> viewSignature s) signatures)
+
+                        SignatureInline ( _, signature ) ->
+                            span [ class "signature-inline" ] [ viewSignature signature ]
+
+                        Eval source result ->
+                            div
+                                [ class "eval" ]
+                                [ UI.codeBlock [] (div [] [ viewSyntax source, div [ class "result" ] [ viewSyntax result ] ]) ]
+
+                        EvalInline source result ->
+                            span [ class "eval-inline" ]
+                                [ UI.inlineCode [] (span [] [ viewSyntax source, span [ class "result" ] [ viewSyntax result ] ]) ]
+
+                        Embed syntax ->
+                            div [ class "embed" ] [ UI.codeBlock [] (viewSyntax syntax) ]
+
+                        EmbedInline syntax ->
+                            span [ class "embed-inline" ] [ UI.inlineCode [] (viewSyntax syntax) ]
+
+                Join docs ->
+                    div [] (List.map viewAtCurrentSectionLevel docs)
+
+                UntitledSection content ->
+                    section [] (List.map viewAtCurrentSectionLevel content)
+
+                Column content ->
+                    div [ class "column" ] (List.map viewAtCurrentSectionLevel content)
+
+                Group content ->
+                    span [ class "group" ] [ viewAtCurrentSectionLevel content ]
+    in
+    div [ class "doc" ] [ view_ 0 document ]
```

Comment:
Should this be `view_ 1 document`? Since when you encounter a section you emit `hN` where `N` is the section level.

Reply:
yeah you're right—good catch!